spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/SparkSchemaUtil.java

```diff
@@ -59,9 +59,7 @@ private SparkSchemaUtil() {}
    * @return a Schema for the table, if found
    */
   public static Schema schemaForTable(SparkSession spark, String name) {
-    StructType sparkType = spark.table(name).schema();
-    Type converted = SparkTypeVisitor.visit(sparkType, new SparkTypeToType(sparkType));
-    return new Schema(converted.asNestedType().asStructType().fields());
+    return convert(spark.table(name).schema());
```

Comment:
I'll just raise this separately since it's a small refactoring not directly related to this change

Reply:
https://github.com/apache/iceberg/pull/10137